patches/woocommerce.0006.fix.attribute-count-cache.patch

```diff
@@ -0,0 +1,41 @@
+From 90ff6c2778657b4ac3e16432aee68c8d71a7106e Mon Sep 17 00:00:00 2001
+From: Fabian Marz <fabian@netzstrategen.com>
+Date: Thu, 16 Jan 2025 23:19:46 +0100
+Subject: [PATCH] perf: add caching to attribute count queries
+
+---
+ .../StoreApi/Utilities/ProductQueryFilters.php | 18 ++++++++++++++++--
+ 1 file changed, 16 insertions(+), 2 deletions(-)
+
+diff --git a/packages/woocommerce-blocks/src/StoreApi/Utilities/ProductQueryFilters.php b/packages/woocommerce-blocks/src/StoreApi/Utilities/ProductQueryFilters.php
+index 22c361eb7..00bc89c46 100644
+--- a/packages/woocommerce-blocks/src/StoreApi/Utilities/ProductQueryFilters.php
++++ b/packages/woocommerce-blocks/src/StoreApi/Utilities/ProductQueryFilters.php
+@@ -134,8 +134,22 @@ class ProductQueryFilters {
+ 		$query_args                   = $product_query->prepare_objects_query( $request );
+ 		$query_args['no_found_rows']  = true;
+ 		$query_args['posts_per_page'] = -1;
+-		$query                        = new \WP_Query();
+-		$result                       = $query->query( $query_args );
++		$cache_key                    = md5( serialize( $query_args ) );
++		$result                       = wp_cache_get( $cache_key, 'counts' );
++
++		if ( !$result ) {
```

Comment:
I'm sorry, this is obviously not correct. I assumed the subject will be taken from the PR title and did not check once I created the PR. I adjusted both subjects in the separate files. While using the StoreAPI and trying to get the attribute counts I noticed the same query ran a couple of times which resulted in a slow down. This patch will add some minor caching which took some inspiration from a rejected/closed PR as seen here: https://github.com/woocommerce/woocommerce/pull/43772/files#diff-66ab9a4c0f13059a5f3f9baecaacb17c509f5f7895a0f1e1b5dfd6d499fdc336R207-R255

Checking the current WooCommerce version, this behaviour still exists. I also opened a related issue here back in January: https://github.com/woocommerce/woocommerce/issues/54265.